### PR TITLE
MINIFICPP-2259 ProcessContext::getProperty should operate on raw poin…

### PR DESCRIPTION
**Pull request by martinzink**
…ters

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/aws/processors/FetchS3Object.h [hunk 1]
```diff
@@ -46,6 +46,7 @@ class FetchS3Object : public S3Processor {
       .supportsExpressionLanguage(true)
       .build();
   EXTENSIONAPI static constexpr auto Version = core::PropertyDefinitionBuilder<>::createProperty("Version")
+      .withDescription("The Version of the Object to download")
```
Comment (szaszm) [inline extensions/aws/processors/FetchS3Object.h:49]: duplicated line

File: libminifi/src/core/ProcessSession.cpp [hunk 2]
```diff
@@ -199,21 +198,19 @@ void ProcessSession::remove(const std::shared_ptr<core::FlowFile> &flow) {
...
 }
 
-void ProcessSession::putAttribute(const std::shared_ptr<core::FlowFile>& flow, std::string_view key, const std::string& value) {
-  flow->setAttribute(key, value);
-  std::stringstream details;
-  details << process_context_->getProcessorNode()->getName() << " modify flow record " << flow->getUUIDStr() << " attribute " << key << ":" << value;
-  provenance_report_->modifyAttributes(flow, details.str());
+void ProcessSession::putAttribute(core::FlowFile& flow_file, std::string_view key, const std::string& value) {
+  flow_file.setAttribute(key, value);
+  std::string details = fmt::format("{} modify flow record {} attribute {} : {}", process_context_->getProcessorNode()->getName(), flow_file.getUUIDStr(), key, value);
```
Comment (szaszm) [inline libminifi/src/core/ProcessSession.cpp:206]: there was no space between the key and the value before the change
```suggestion
  std::string details = fmt::format("{} modify flow record {} attribute {}:{}", process_context_->getProcessorNode()->getName(), flow_file.getUUIDStr(), key, value);
```

File: extensions/aws/processors/FetchS3Object.h (hunk 1 above)
Comment (martinzink) [reply extensions/aws/processors/FetchS3Object.h]: good catch fixed it in https://github.com/apache/nifi-minifi-cpp/pull/1707/commits/d64e7f890b82be04ec5a75bc484a1ebcbb001f32

File: libminifi/src/core/ProcessSession.cpp (hunk 2 above)
Comment (martinzink) [reply libminifi/src/core/ProcessSession.cpp]: good catch fixed it in https://github.com/apache/nifi-minifi-cpp/pull/1707/commits/d64e7f890b82be04ec5a75bc484a1ebcbb001f32

**Review by szaszm — APPROVED** [review]

File: libminifi/src/core/ProcessSession.cpp [hunk 3]
```diff
@@ -110,36 +109,36 @@ std::shared_ptr<core::FlowFile> ProcessSession::create(const std::shared_ptr<cor
     }
     record->setLineageStartDate(parent->getlineageStartDate());
     record->setLineageIdentifiers(parent->getlineageIdentifiers());
-    parent->getlineageIdentifiers().push_back(parent->getUUID());
+    record->getlineageIdentifiers().push_back(parent->getUUID());
```
Comment (fgerlits) [inline libminifi/src/core/ProcessSession.cpp:112]: was this a bug?

Comment (martinzink) [reply libminifi/src/core/ProcessSession.cpp:112]: yeah 🐛

**Review by adamdebreceni — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**